doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Linux/index.md

```diff
@@ -4,302 +4,300 @@ title: Linux
 permalink: /:path/
 sort_idx: 10
 ---
+{% assign root_url = page.url | split: '/'%}
+{% capture root_url  %} /{{root_url[1]}}/{{root_url[2]}}/{% endcapture %}
+
+- [Installing prerequisites](#installing-prerequisites)
+- [Compile C++ endpoint SDK](#compile-c-endpoint-sdk-1)
+- [Configure C++ endpoint SDK](#configure-c-endpoint-sdk-1)
+- [Quick way to build SDK](#quick-way-to-build-sdk)
+    - [Build in Docker container](#build-in-docker-container)
+    - [Build in nix shell](#build-in-nix-shell)
+- [Minimal example](#minimal-example)
+    - [Directory structure](#directory-structure)
+    - [Build system overview](#build-system-overview)
+    - [Code](#code)
+    - [Buliding](#building)
+- [What's next?](#whats-next)
+
+This page describes Kaa C++ SDK build process on Linux system.
 
-Use the following instructions to build endpoint SDKs in Java, C, and C++ for Linux.
+**Verified against:**
 
-<!-- START doctoc generated TOC please keep comment here to allow auto update -->
-<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Host OS:** Ubuntu 14.04 LTS Desktop 64-bit.
 
+## Installing prerequisites
 
-- [Java endpoint SDK](#java-endpoint-sdk)
-- [C endpoint SDK](#c-endpoint-sdk)
-    - [Install build utils and dependencies](#install-build-utils-and-dependencies)
-    - [Compile C endpoint SDK](#compile-c-endpoint-sdk)
-    - [Configure C endpoint SDK](#configure-c-endpoint-sdk)
-- [C++ endpoint SDK](#c-endpoint-sdk-1)
-    - [Install build utils and dependencies](#install-build-utils-and-dependencies)
-    - [Compile C++ endpoint SDK](#compile-c-endpoint-sdk-1)
-    - [Configure C++ endpoint SDK](#configure-c-endpoint-sdk-1)
-- [Quick way to build C/C++ endpoint SDK](#quick-way-to-build-c-c-endpoint-sdk)
+Before building the C++ endpoint SDK, install the following components on your machine:
 
-<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+1. Install compiler:
+    1. For automatic installation, execute the following commands:
 
-**Verified against:**
+       ``` shell
+       sudo apt-get install g++
+       ```
 
-**Host OS:** Ubuntu 14.04 LTS Desktop 64-bit.
+    2. For manual installation of version 4.8, refer to the following example:
 
-# Java endpoint SDK
+       ``` shell
+       sudo apt-get install python-software-properties
+       sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+       sudo apt-get update
+       sudo apt-get install g++-4.8
+       sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 50
+       ```
```

Comment:
I vote for removing this section. `sudo apt-get install g++` is superior for user experience.